root/io/TFile/execNestedCollectionStreamerInfo.cxx

```diff
@@ -0,0 +1,50 @@
+#include <vector>
+#include "TTree.h"
+#include "TFile.h"
+#include "TMemFile.h"
+#include <iostream>
+#include <memory>
+
+struct Inside {
+  int fValue;
+};
+
+struct Middle {
+  std::vector<Inside> fInside;
+};
+
+struct Outer {
+  std::vector<Middle> fMiddle;
+};
+
+bool CheckNestedCollectionStreamerInfo(bool empty = true, const char *filename = "forceinfo.root")
```

Comment:
Could you add a comment why `empty` and `!empty` might give different results? And how this tests #11436 ?

Reply:
```
   // Test both empty and filled collections as the StreamerInfo handling is different
   // in some cases (split vs non-split, object-wise streaming vs member-wise streaming).
   // In the case of the original issue (https://github.com/root-project/root/issues/11436)
   // having a filled collection did not help.
```